pipenv/cli/options.py

```diff
@@ -300,15 +318,32 @@ def callback(ctx, param, value):
                   help="Import a requirements.txt file.", callback=callback)(f)
 
 
-def requirements_flag(f):
+def emit_requirements_flag(f):
     def callback(ctx, param, value):
         state = ctx.ensure_object(State)
         if value:
-            state.installstate.requirementstxt = value
+            state.lockoptions.emit_requirements = value
         return value
     return option("--requirements", "-r", default=False, is_flag=True, expose_value=False,
                   help="Generate output in requirements.txt format.", callback=callback)(f)
 
+def emit_requirements_header_flag(f):
+    def callback(ctx, param, value):
+        state = ctx.ensure_object(State)
+        if value:
+            state.lockoptions.emit_requirements_header = value
+        return value
+    return option("--header/--no-header", default=True, is_flag=True, expose_value=False,
+                  help="Add header to generated requirements", callback=callback)(f)
+
+def dev_only_flag(f):
```

Comment:
I normally try to avoid mixing cosmetic changes with functional ones, but in this case, I couldn't follow my own code without making the cosmetic change first :)